packages/@jsii/check-node/src/constants.ts

```diff
@@ -1,69 +1,161 @@
+          knownBroken: !nodeRelease.supportedRange.test(semver),
+        };
+      }
+    }
+
+    return { nodeRelease: undefined, knownBroken: false };
+  }
 
   /**
-   * Not end-of-life, but also not supported. This is usually the case for early
-   * releases on a new major train, where some of the features are not available
-   * yet. These emit a SCARY RED WARNING, but the messaging is slightly
-   * different than that of END_OF_LIFE.
+   * The major version of this node release.
    */
-  UNSUPPORTED = 'unsupported',
+  public readonly majorVersion: number;
 
   /**
-   * Those releases are newer than the current software package, so we have not
-   * tested with it. People may encounter all kinds of weird bugs, especially
-   * early in the release cycle... So we'll issue a yellow warning here.
+   * The date on which this release range starts to be considered end-of-life.
+   * Can also be `true` to denote that this is already end-of-life, without
+   * specifying a particular date.
    */
-  UNTESTED = 'untested',
-}
+  public readonly endOfLifeDate: Date | undefined;
 
-/**
- * The deadline after which deprecated releases will move to
- * `SupportLevel.END_OF_LIFE`.
- */
-export const DEADLINE = '2021-09-01';
-/**
- * The DEADLINE, expressed in milliseconds since the epoch.
- */
-export const DEADLINE_EPOCH_MS = new Date(
-  `${DEADLINE}T00:00:00.000Z`,
-).getTime();
+  /**
+   * Determines whether this release is currently considered end-of-life.
+   */
+  public readonly endOfLife: boolean;
 
-/**
- * For each SemVer range, what is the support level for such versions of Node.
- * The ranges are checked in declaration order (so it is a good idea to keep the
- * left-open ranges at the beginning, and the right-open ranges at the end.
- * First match wins.
- *
- * Also, if you are setting a new entry to `SupportLevel.DEPRECATED`, do not
- * forget to also change the `DEADLINE` value.
- */
-export const VERSION_SUPPORT: { readonly [range: string]: SupportLevel } = {
-  '<12.0.0-0': SupportLevel.END_OF_LIFE,
-  '<12.7.0': SupportLevel.UNSUPPORTED,
-  '^12.7.0': SupportLevel.SUPPORTED,
-  '^13.0.0-0': SupportLevel.END_OF_LIFE,
-  '<14.5.0': SupportLevel.UNSUPPORTED,
-  '^14.5.0': SupportLevel.SUPPORTED,
-  '^15.0.0-0': SupportLevel.END_OF_LIFE,
-  '<16.3.0': SupportLevel.UNSUPPORTED,
-  '^16.3.0': SupportLevel.SUPPORTED,
-  '<17.3.0': SupportLevel.UNSUPPORTED,
-  '^17.3.0': SupportLevel.SUPPORTED,
-  // Anything else will be treated as SupportLevel.UNTESTED.
-};
+  /**
+   * Determines whether this release is within the deprecation window ahead of
+   * it's end-of-life date.
+   */
+  public readonly deprecated: boolean;
+
+  /**
+   * If `true` denotes that this version of node has not been added to the test
+   * matrix yet. This is used when adding not-yet-released versions of node that
+   * are already planned (typically one or two years out).
+   *
+   * @default false
+   */
+  public readonly pending: boolean;
```

Comment:
(optional suggestion)
```suggestion
  public readonly notReleasedYet: boolean;
```

Reply:
Renamed to `untested` which is a more accurate description of the intent.